lumium-space/src/security/authentication/authenticators/Authenticator.tsx

```diff
@@ -1,11 +1,26 @@
-import ThirdPartyEmailPassword from 'supertokens-auth-react/recipe/thirdpartyemailpassword'
-import dynamic from 'next/dynamic'
+import { useEffect } from 'react';
+import Router, { useRouter } from 'next/router';
+import { useLoginStatus } from "@hooks";
+import React from 'react';
 
-const ThirdPartyEmailPasswordAuthNoSSR = dynamic(
-    new Promise((res) =>
-        res(ThirdPartyEmailPassword.ThirdPartyEmailPasswordAuth)
-    ) as any,
-    { ssr: false }
-)
-
-export { ThirdPartyEmailPasswordAuthNoSSR as Authenticator };
+export const Authenticator: React.FC = (props) => {
+    const {
+        children,
+    } = props;
+    const { asPath } = useRouter();
+    const loggedIn = useLoginStatus();
+    useEffect(() => {
+        const path = "/auth/signin?redirectionURL=" + encodeURIComponent(asPath.substring(0, asPath.lastIndexOf("/")));
+        if (!loggedIn) {
+            if (asPath.includes("auth")) {
```

Comment:
i think it would be nice if we could be a bit more careful here, `includes` is probably too broad. other than that, good work on this authenticator component :+1:

Reply:
I don't really get what you mean by that. Reason behind this is that it shouldnt redirect to an url starting with /auth

Reply:
i'm trying to say that we should check the path of the route with next/router. otherwise, if a visitor adds something like `?auth` at the end then this `includes` would also return true which could be a potential security issue. it is definitely a code smell and should be removed